S7.Net/PlcAsynchronous.cs

```diff
@@ -54,15 +78,12 @@ public async Task OpenAsync(CancellationToken cancellationToken = default)
             if (s7data.Length < 20)
                 throw new WrongNumberOfBytesException("Not enough data received in response to Communication Setup");
 
-            MaxPDUSize = (short)(s7data[18] * 256 + s7data[19]);
-        }
-
-        private async Task ConnectAsync()
-        {
-            tcpClient = new TcpClient();
-            ConfigureConnection();
-            await tcpClient.ConnectAsync(IP, Port);
-            _stream = tcpClient.GetStream();
+            // TODO: check if this should not rather be UInt16.
+            MaxPDUSize = Types.Int.FromByteArray(s7data, 18);
+            if (MaxPDUSize <= 0 )
```

Comment:
[RFC905](https://tools.ietf.org/html/rfc905) 6.5.4 j status that the minimum is actually 128, which also is the maximum length for the CR TPDU. Also, this number is unsigned and would logically always be unsigned in any protocol, so the < 0 part would most logically be invalid interpretation on our side 😜.

Reply:
Well unless we figure out how exactly the PDU should be parsed, I am not adding a check. But we should remember this and once we are absolutely sure we can also check this lower limit.